third_party/proto/gogoproto/gogo.proto

```diff
@@ -0,0 +1,145 @@
+// Protocol Buffers for Go with Gadgets
```

Comment:
I have no idea. It is referenced in our proto files, and I needed to include it to get linting to work.

Reply:
I'm pretty sure we don't need to include gogogproto here. It comes from the gogo proto compiler.
Maybe add a buf.yaml file here similar to https://github.com/cosmos/cosmos-sdk/blob/main/third_party/proto/buf.yaml

Reply:
Pretty sure that these are referencing this file : 
https://github.com/cosmos/cosmos-sdk/blob/main/proto/cosmos/base/v1beta1/coin.proto

We probably shouldn't copy these into this repo since then we'd have to maintain it

Reply:
I've tried several solutions to this, and nothing works. I've tried the deps solution but buf then needs a buf.lock file. I make one, and then buf breaks because the branch isn't main. It's a known bug in buf. Buf works if there is no buf.lock file. 

I've tried changing around the deps paths to no avail either. 

At this point, leaving it there to get the linter to work is the best solution. Or, leave the linter broken.

Reply:
All right @adamewozniak and @robert-zaremba - I rewrote the Makefile. This is a working solution that won't force us to maintain anything. Using deps did not work sadly.

Reply:
Works for me - does that mean we can remove the files now @brianosaurus ?
Would love @robert-zaremba to look at this again as well btw

Reply:
Works for me - does that mean we can remove the files now @brianosaurus ?
Would love @robert-zaremba to look at this again as well btw

Reply:
I checked a couple of cosmos projects ([osmosis](https://github.com/osmosis-labs/osmosis/tree/main/proto/osmosis), [persistence](https://github.com/persistenceOne/persistenceCore/tree/master/proto/persistence/halving/v1beta1), [regen](https://github.com/regen-network/regen-ledger/tree/master/proto)) and didn't find any that maintained copies of the cosmos sdk proto files.

I think the closest makefile to what we want is [the one from regen](https://github.com/regen-network/regen-ledger/blob/master/Makefile#L346) The others didn't check proto as strongly.

I'd strongly prefer a solution that doesn't add a bunch of files with rare, but potential, opportunities for us to fall out of sync when sdk updates its proto.

Reply:
@toteki and @adamewozniak They get refreshed every time someone builds proto-all from the respective SDK. They wouldn't fall out of sync. We aren't maintaining any of the files. However, I investigated .gitignore and determined that, yes, we can just .gitignore these. It's a good suggestion. I'm curious why anyone else hasn't made the same choice?

In any case, I've modified the clang-formatter to properly format our *.proto files (the ones in the proto directory), I removed the third_party directory from the repo, and added third_party to .gitignore.

Reply:
I had to change the workflow to get it to work in google workflows. Building clean now.